api/v1beta1/types.go

```diff
@@ -92,8 +92,10 @@ type Subnet struct {
 // VPCEndpoint describes a VPCEndpoint.
 type VPCEndpoint struct {
 	Address *string `json:"address"`
-	// +optional
+
 	// Deprecated: This field has no function and is going to be removed in the next release.
+
+	// +optional
```

Comment:
Can you follow something like this https://github.com/openshift-cloud-team/cluster-api/blob/ee2b6e857aa21f0a94c80024f60bf558a4a151a7/api/core/v1beta1/cluster_types.go#L564-L570

It will preserve the Deprecated fields in the CRD.